### PR TITLE
Log to file and change log level on demand

### PR DESCRIPTION
**Pull request by shackra**
Modify the `Manager` interface with 2 new methods:
- `ShouldDebug()` tells if the `connectorRunner` should debug to a file and switch the log level to `debug`.
- `GetTempDir()` returns the current path we should create the logging file, this happens to be the same as the `c1z` file location.

A new gRPC message was also added, it is called `StartDebugging`.

## Summary by CodeRabbit

- **New Features**
	- Introduced new debugging capabilities, including a `StartDebugging` RPC method and associated request/response structures.
	- Enhanced validation logic for new message types improving error handling and feedback.
	- Added methods for enabling debugging and retrieving temporary directory paths across various managing components.
  
- **Bug Fixes**
	- Improved error handling in the validation process for debugging-related requests.

- **Documentation**
	- Added comments to the codebase to clarify functionalities and improve maintainability.

- **Chores**
	- Expanded interface definitions to support new methods related to debugging for various task management components.

### CODE REVIEW

File: pkg/tasks/c1api/full_sync.go [hunk 1]
```diff
@@ -30,14 +30,16 @@ type fullSyncTaskHandler struct {
...
+func (c *fullSyncTaskHandler) sync(ctx context.Context, c1zPath string, debug bool) error {
 	l := ctxzap.Extract(ctx).With(zap.String("task_id", c.task.GetId()), zap.Stringer("task_type", tasks.GetType(c.task)))
 	syncer, err := sdkSync.NewSyncer(ctx, c.helpers.ConnectorClient(), sdkSync.WithC1ZPath(c1zPath), sdkSync.WithTmpDir(c.helpers.TempDir()))
 	if err != nil {
 		l.Error("failed to create syncer", zap.Error(err))
 		return err
 	}
 
+	// TODO(shackra): Change log-level to debug if `debug` is true
+
```
Comment (ggreer) [inline pkg/tasks/c1api/full_sync.go:42]: Also todo: figure out how to tell the logger to save its logs to a file next to the temp c1z (initialized right above).

**Comment by shackra** [comment]
@coderabbitai review

surprise me!

**Comment by shackra** [comment]
> Tip
> 
> For best results, initiate chat on the files or code changes.
> 
> @shackra: Here is a detailed review of the changes:
> ### `pkg/sync/syncer.go`
> 
>     * The changes in this file are minor and involve formatting adjustments to the `Sync` and `Close` method signatures. These changes are straightforward and improve code readability.
> 
> 
> ### `pkg/tasks/c1api/full_sync.go`
>
>     * These changes are well-implemented and significantly improve the debugging capabilities of the synchronization process.
> 
> 
> Overall, the changes are well-implemented and enhance the debugging capabilities of the synchronization process. Great job!

oh stop it, you

File: pkg/connectorrunner/runner.go [hunk 2]
```diff
@@ -42,6 +45,28 @@ func (c *connectorRunner) Run(ctx context.Context) error {
...
+	if c.tasks.ShouldDebug() && c.debugFile == nil {
+		var err error
+		c.debugFile, err = os.Create(filepath.Join(c.tasks.GetTempDir(), "debug.log"))
+		if err != nil {
+			return err
+		}
+	}
+
+	// modify the context to insert a logger directed to a file
+	if c.debugFile != nil {
```
Comment (mgaeta) [inline pkg/connectorrunner/runner.go:57]: At this point, `c.debugFile` cannot be nil.

**Review by mgaeta — APPROVED** [review]

**Pull request merged**